Mathlib/Order/UpperLower/Basic.lean

```diff
@@ -946,6 +983,24 @@ theorem compl_iInf₂ (f : ∀ i, κ i → LowerSet α) :
     (⨅ (i) (j), f i j).compl = ⨅ (i) (j), (f i j).compl := by simp_rw [LowerSet.compl_iInf]
 #align lower_set.compl_infi₂ LowerSet.compl_iInf₂
 
+-- TODO: Introduce an `IsMaxOn` predicate?
+/-- Remove a maximal element from a lower set. -/
+def erase (s : LowerSet α) (a : α) (has : ∀ b ∈ s, a ≤ b → b = a) : LowerSet α where
+  carrier := s \ {a}
+  lower' := s.lower.erase has
+
+lemma erase_le {has} : s.erase a has ≤ s :=diff_subset _ _
+
+lemma erase_lt {has} (ha : a ∈ s) : s.erase a has < s :=
+  sdiff_lt (singleton_subset_iff.2 ha) $ singleton_ne_empty _
+
+@[simp, norm_cast]
+lemma coe_erase (s : LowerSet α) (a : α) (ha) : (s.erase a ha : Set α) = ↑s \ {a} := rfl
```

Comment:
Is the arrow on the rhs necessary here?

Reply:
Yes. Both sides of an equality are npw elaborated separately.